pkg/rke2/psp/psp.go

```diff
@@ -0,0 +1,443 @@
+				}
+			}
+		}
+
+		ns.SetAnnotations(map[string]string{namespaceAnnotationGlobalUnrestricted: cisAnnotationValue})
+	}
+	return nil
+}
+
+// setSystemUnrestricted sets the system unrestricted podSecurityPolicy as
+// the associated role and rolebinding.
+func setSystemUnrestricted(cs *kubernetes.Clientset, ns *v1.Namespace) error {
+	if _, ok := ns.Annotations[namespaceAnnotationSystemUnrestricted]; !ok {
+		if _, err := cs.RbacV1().ClusterRoles().Get(context.TODO(), systemUnrestrictedPSPName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(systemUnrestrictedPSPTemplate, systemUnrestrictedPSPName)
+				if err := deployPodSecurityPolicyFromYaml(cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoles().Get(context.TODO(), systemUnrestrictedRoleName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(roleTemplate, systemUnrestrictedRoleName, systemUnrestrictedPSPName)
+				if err := deployClusterRoleFromYaml(cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(context.TODO(), systemUnrestrictedRoleBindingName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(systemUnrestrictedNodesRoleBindingTemplate, systemUnrestrictedRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployClusterRoleBindingFromYaml(cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(context.TODO(), systemUnrestrictedSvcAcctRoleBindingName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				tmpl := fmt.Sprintf(systemUnrestrictedServiceAcctRoleBindingTemplate, systemUnrestrictedSvcAcctRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployRoleBindingFromYaml(cs, tmpl); err != nil {
+					return err
+				}
+			}
+		}
+		ns.SetAnnotations(map[string]string{namespaceAnnotationSystemUnrestricted: cisAnnotationValue})
+	}
+	return nil
+}
+
+// kubeConfigExists checks if the kubeconfig file
+// has been written.
+func kubeConfigExists(kc string) bool {
+	info, err := os.Stat(kc)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// SetPSPs sets the default PSP's based on the mode that RKE2 is running in. There is either CIS or non
+// CIS mode. For CIS mode, the globalRestricted and systemUnrestricted polices and associated roles and
+// bindings are created. For non CIS mode, globalRestricted, system-unrestricted, and global-unrestricted
+// policies and associated roles and bindings are loaded, if applicable.
+//
+// Each PSP has an annotation associated with it that we check to see exists before performing any
+// write operations. The annotation is created upon successful setting of PSPs.
+//
+// Load logic
+//
+// CIS:
+// - Check if the globalRestricted annotation does not exist, create PSP, role, binding.
```

Comment:
Very pedantic nitpick: This reads oddly; the "check" is throwing me off.
It would be a little easier to read for me if it followed if/then, like: "If the globalRestricted annotation does not exist, then create PSP, role, & binding."

This is just my opinion; if you like it as-is, don't feel obliged to change it.